diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -392,18 +395,36 @@ CallbackReturn DiffDriveController::on_configure(const rclcpp_lifecycle::State &
   previous_commands_.emplace(*received_velocity_msg_ptr_);
 
   // initialize command subscriber
-  velocity_command_subscriber_ = lifecycle_node_->create_subscription<Twist>(
-    DEFAULT_COMMAND_TOPIC, rclcpp::SystemDefaultsQoS(), [this](
-      const std::shared_ptr<Twist> msg) -> void {
-      if (!subscriber_is_active_) {
-        RCLCPP_WARN(
-          lifecycle_node_->get_logger(),
-          "Can't accept new commands. subscriber is inactive");
-        return;
-      }
+  if (use_stamped_vel_) {
+    velocity_command_subscriber_ = lifecycle_node_->create_subscription<Twist>(
+      DEFAULT_COMMAND_TOPIC, rclcpp::SystemDefaultsQoS(), [this](
+        const std::shared_ptr<Twist> msg) -> void {
+        if (!subscriber_is_active_) {
+          RCLCPP_WARN(
+            lifecycle_node_->get_logger(),
+            "Can't accept new commands. subscriber is inactive");
+          return;
+        }
+
+        received_velocity_msg_ptr_ = std::move(msg);
+      });
+  } else {
+    velocity_command_unstamped_subscriber_ =
+      lifecycle_node_->create_subscription<geometry_msgs::msg::Twist>(
+      DEFAULT_COMMAND_TOPIC, rclcpp::SystemDefaultsQoS(), [this](
```

Comment:
It is not clear to me if you want to use the same topic regardless of the message type or this is a typo and should be DEFAULT_COMMAND_UNSTAMPED_TOPIC` instead.